scifight/templatetags/scifight_url.py

```diff
@@ -0,0 +1,53 @@
+from django import template
+from scifight import models
+from django.core.urlresolvers import reverse, NoReverseMatch
+
+
+register = template.Library()
+
+_single_arg_views = {
+    models.Participant: ("scifight:participant",   'id'),
+    models.Leader:      ("scifight:leader",        'id'),
+    models.Jury:        ("scifight:jury",          'id'),
+    models.Room:        ("scifight:room",          'id'),
+    models.Fight:       ("scifight:fight",         'id'),
+    models.Problem:     ("scifight:problem",       'problem_num')
+}
+
+
+@register.filter(is_safe=True)
+def scifight_url(model):
+    """
+    This filter simplifies the use of a template.
+    Input parameter is a model object.
+    {{ object | scifight_url }}
+    The filter will return the URL, from the corresponding
+    item in the urls.py. In case of an error NoReverseMatch
+    returns an empty string
+    """
+    def wrapped_reverse(*args, **kwargs):
+        try:
+            return reverse(*args, **kwargs)
+        except NoReverseMatch:
+            return ''
+
+    if isinstance(model, models.Team):
+        if model.slug:
+            kwargs = dict(team_slug=model.slug,
+                          tournament_slug= model.tournament.slug)
+            return wrapped_reverse('scifight:team_slug', kwargs=kwargs)
+        else:
+            kwargs = dict(team_id=model.id,
+                          tournament_slug=model.tournament.slug)
+            return wrapped_reverse('scifight:team_id', kwargs=kwargs)
+
+    if isinstance(model, models.Tournament):
+        return wrapped_reverse('scifight:tournament', args=(model.slug,))
+
+    model_type = type(model)
+    if model_type in _single_arg_views:
+        (view_name, field_name) = _single_arg_views[model_type]
+        args = (model.tournament.slug, getattr(model, field_name))
+        return wrapped_reverse(view_name, args=args)
+
+    return ''
```

Comment:
Please, add also error logging before returning empty string. It's pretty bad to have absolutely no error reporting if reverse resolve failed.